src/Microsoft.AspNet.Mvc.ModelBinding/Metadata/DataMemberRequiredBindingMetadataProvider.cs

```diff
@@ -29,7 +29,7 @@ public void GetBindingMetadata([NotNull] BindingMetadataProviderContext context)
             }
 
             var dataMemberAttribute = context
-                .Attributes
+                .PropertyAttributes
```

Comment:
This is slightly more correct than what we were doing here before (which relies on the fact that `[DataMember]` can't be put on a type).